virtualmachine: Add support for Disk Encryption Set parameter

lm-sig opened the pull request:
##### SUMMARY
I had a requirement to have virtual machines with disks (both OS and data) that used encryption. The VM module did not support provisioning or attaching any disk type with encryption enabled.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
Fixes #425

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
azure_vm_virtualmachine

##### ADDITIONAL INFORMATION
A customer may set an Azure policy that requires VMs to have disks with encryption. If you attempt to provision a virtual machine with disks without encryption your request will fail.

This change will support both encrypted or unencrypted disks. Currently it requires you to provide the full Azure resource ID for the data encryption set ID.

lm-sig committed:
virtualmachine: Add support for Disk Encryption Set parameter

lm-sig committed:
virtualmachine: Add support for Disk Encryption Set parameter

Fred-sun commented on plugins/modules/azure_rm_virtualmachine.py:
```diff
@@ -203,6 +203,15 @@
+    os_disk_encryption_set:
+        description:
+            - Disk encryption set for the OS disk.
+        type: dict
```
Consistent with the definition of arg
```suggestion
        type: raw
```

lm-sig replied:
Updated. Thanks.

Fred-sun reviewed (commented):
line 1634, 1810 and 2066 line too long,  longer than 160 characters.

lm-sig committed:
virtualmachine: Address pull request comments regarding type and line length

lm-sig committed:
virtualmachine: Address pull request comments regarding type and line length

Rickmarges committed:
Add creation timestamp to hostvars (#1221)

Co-authored-by: Marges, RSY (Rick) <rick.marges@achmea.nl>

timway committed:
Improve Docs In `azure_rm_virtualnetwork` (#1203)

* Improve Docs In `azure_rm_virtualnetwork`

* The DNS Servers should be a list of strings
* Remove the trailing comma in the `dict()` call for `dns_servers`
* The boolean values should default to a boolean value in the docs
* Clean up excess quotation in some of the examples

* Set the Argument Spec up to Validate List Element Typing

* Incorporate PR feedback from @Fred-sun correctly indicated that I didn't apply the elements parameter to the argument spec for `address_prefixes_cidr` and `dns_servers`
* This helps Ansible validate the list elements are the correct type for the 2 affected module parameters

* Remove Unneeded Ignores In Sanity

* As @Fred-sun pointed out a number of the sanity tests no longer need to ignore a failure condition with this PR

Fred-sun committed:
Add new parameters to storageshare (#1216)

* Add new parameters to azure_rm_storageshare.py

* Add test case

* Fix sanity fail

* small change

* small change 02

* remove test case

* small change

Klaas- committed:
Add an example on how to add a domain to hostname for dynamic inventory (#1211)

cpiment committed:
fix typo update_security_profle (#1194)

Changed update_security_profle for update_security_profile

xuzhang3 committed:
Fix `azure_rm_backupazurevm` -  Get module defined  `subscrtion_id` (#1225)

* use module sub_id if definied

* use module sub_id if definied

* lint

Fred-sun committed:
Migrate ADAL to MSAL (#1239)

* Migrate ADAL to MSAL

* Fix the error when the password is empty

* sanity change

* Fix sanity fail

* small change

* Delete unused parameters

* delete x509_private_key_path

dependabot[bot] committed:
Bump cryptography from 38.0.3 to 39.0.1 (#1076)

Bumps [cryptography](https://github.com/pyca/cryptography) from 38.0.3 to 39.0.1.
- [Release notes](https://github.com/pyca/cryptography/releases)
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/38.0.3...39.0.1)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cryptography from 39.0.1 to 41.0.3 (#1244)

Bumps [cryptography](https://github.com/pyca/cryptography) from 39.0.1 to 41.0.3.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/39.0.1...41.0.3)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Fred-sun committed:
Upgrade azure mgmt batch 17.0.0 (#1202)

* add azure_rm_batchaccount_info.py

* Improve azure_rm_batchaccount test case

* change version to 16.0.0

* change version to 15.0.0

* new change

* last change

* small change

* last change

Fred-sun committed:
Fix azure_rm_aks_info.py to list all manged cluster bug (#1229)

msl0 committed:
removed subscription_id from azure_keyvault_secret lookup when using az cli auth (#1175)

* Update azure_keyvault_secret.py

Added Azure CLI support
Updated documentation

* Update azure_keyvault_secret.py

Removed subscription_id from examples

* Applied suggestions

* import order changed

Fred-sun committed:
Bump version to v1.17.0 (#1247)

* Bump version to v1.17.0

* update change log

* update change log

* Add new commits

---------

Co-authored-by: xuzhang3 <Zhangxu894765>

rndmh3ro committed:
Fix typo (#1254)

Fred-sun committed:
Migrate msrest to azure-core (#1245)

* Migrate msrest to azure-core

* Fix missing change

* fix sanity fail

* Fixed import failure

* restore ignore file

* Fix missing change

* Update msrestazure'method to azure-mgmt-core'method

* Updates how http request return values are handled

* new change

* Changes the handling of return values

* Update azure_rm_appgateway test case

* Update azure_rm_appgateway test case

* Update azure_rm_iotdevice* to support trunck2

* update test case

* Add new change

* new change

* fix systax error:

* Modify azure_rm_virtualmachine.py

* Modify azure_rm_virtualmachinescalset.py

* Modify azure_rm_virtuamachinescalset.py

* last changed

* fix test fail

* last change

* Add query parameters to azure_rm_iotdevice_info

xuzhang3 committed:
Lazy initialize the AD resource credentials (#1268)

* lazy init track1 credentials

* fix sanity

xuzhang3 committed:
changelog v1.18.0 (#1269)

dependabot[bot] committed:
Bump cryptography from 41.0.3 to 41.0.4 (#1265)

Bumps [cryptography](https://github.com/pyca/cryptography) from 41.0.3 to 41.0.4.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/41.0.3...41.0.4)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kmj251 committed:
Fix azure_object.type to equal Microsoft.Management/managementGroups. Fixes #1240 (#1252)

Fred-sun committed:
Change ansible default version from 2.14 to 2.15 (#1272)

* Change ansible default version from 2.14 to 2.15

* fix sanity fail

* fix santiy error --- 02

xuzhang3 committed:
init trunk1 credentials (#1271)

xuzhang3 committed:
v1.18.1 changelog (#1273)

lm-sig committed:
azure_rm_virtualmachine: Fix VM update with disk encryption set

lm-sig committed:
azure_rm_virtualmachine: Fix VM update with disk encryption set

lm-sig committed:
virtualmachine: Add support for Disk Encryption Set parameter

# Conflicts:
#	plugins/modules/azure_rm_virtualmachine.py

lm-sig commented:
I updated the commits against the latest 'dev' branch changes. Tested against Azure US Gov cloud. Still creating VMs with encrypted disks.

xuzhang3 commented:
@lm-sig  can you help resolve the conflict.

lm-sig commented:
I am going to open a new pull request since the dev branch keeps changing.
